examples/nlp/data/sft/remove_long_dialogues.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+
+from nemo.collections.nlp.data.language_modeling.megatron.gpt_sft_chat_dataset import GPTSFTChatDataset
+from nemo.collections.nlp.modules.common.tokenizer_utils import get_nmt_tokenizer
+
+"""
+A script for removing problematic long dialogues from a chat dataset prior to training.
+Any example that does not have any non-masked tokens after truncation is removed.
+
+Usage:
+  python3 remove_long_dialogues.py \
+    --tokenizer_file <PATH TO SENTENCEPIECE TOKENIZER MODEL> \
+    --dataset_file <PATH TO DATASET TO PREPROCESS> \
+    --output_file <WHERE TO SAVE PREPROCESSED DATASET> \
+    --seq_len <MAX_SEQ_LEN TO USE DURING TRAINING>
+"""
+
+
+def test_index(tokenizer_file, dataset_file, output_file, seq_len=4096):
+    tokenizer = get_nmt_tokenizer(library="sentencepiece", tokenizer_model=tokenizer_file,)
```

Comment:
does this generalize or can only be used when the tokenizer is sentencepiece?

Reply:
It should generalize. I can update the script to also support HF tokenizers